sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/action/QueryChangeStreamAction.java

```diff
@@ -189,7 +188,8 @@ public ProcessContinuation run(
                     (DataChangeRecord) record,
                     tracker,
                     receiver,
-                    watermarkEstimator);
+                    watermarkEstimator,
```

Comment:
Yes, all the work that is required for the data change records is contained within the corresponding action then. It was in this class before because it applied to all the types of records before, but you can see that we always delegate to the corresponding actions for specific behaviour.

IMHO the logic is better encapsulated.